### PR TITLE
Ramone/verify contract

### PR DESCRIPTION
**Pull request by Ramone-HL**
Added a new section, Verifying a Smart Contract in the Deploy chapter including screenshots.

### CODE REVIEW

File: static/img/docs/deploy/verifyssuccess1.png
Comment (taijonsmith) [inline static/img/docs/deploy/verifyssuccess1.png:1]: update filename to `verifysuccess1.png` to fix the broken build

File: docs/deploy.md [hunk 1]
```diff
@@ -548,3 +548,102 @@ This exposes a JSON-RPC interface to the Hardhat Network. Use the address, **htt
...
+
+**Note:** See the [Build](https://eon.horizen.io/docs/build) section for details on compiling and deploying a smart contract.
+
+* Flatten the smart contract, if needed
+* Check for the deployed contract using the EON Explorer
+* Verify and publish the contract
+
+Make sure that you deploy your contract to the Yuma Testnet. If you are using a wallet, you must connect it to the Yuma Testnet network.
+
+Use the following parameters to set connect to Yuma Testnet:
```
Comment (taijonsmith) [inline docs/deploy.md:570]:
```suggestion
Use the following parameters to connect to Yuma Testnet:
```

File: docs/deploy.md [hunk 2]
```diff
@@ -548,3 +548,102 @@ This exposes a JSON-RPC interface to the Hardhat Network. Use the address, **htt
...
+
+  * **ABI-encoded Constructor Arguments:** See the [ABI Constructor Arguments](https://docs.blockscout.com/for-users/abi-encoded-constructor-arguments) for detailed information. 
+
+  * **Add Contract Libraries:** Enter the **name** and **0x address** of any required libraries called in the **.sol** file. If your contract is  flattened, you do not need to add a contract library.
+
+7. Click **Verify & publish**. A checkmark symbol appears next in the **Code** tab, if successful. Also, the **Read Contract** tab and **Write Contract** tab appears.
+
+  **Note:** It is recommended **NOT** to use the **Brave web browser** for **EON Explorer** contract verification because of interface limitations.
+  
+8. Click **Read Contract** tab or **Write Contract** tab to display and interact with the contract details and any related transactions.
```
Comment (taijonsmith) [inline docs/deploy.md:644]:
```suggestion
8. Click the **Read Contract** tab or **Write Contract** tab to display and interact with the contract details and any related transactions. 
```

File: docs/deploy.md [hunk 3]
```diff
@@ -548,3 +548,102 @@ This exposes a JSON-RPC interface to the Hardhat Network. Use the address, **htt
...
+Before using the EON Explorer, the smart contract you want to verify has been deployed in the Yuma Testnet network. You need to flatten the smart contract if the contract inherits from other contracts or if it uses a library. Flattening a smart contract refers to combining all Solidity code into one file instead of multiple source files, thereby having the imported code be embedded in the same file. 
+
+All smart contract development tools have a source code flattener. For example: 
+
+* **Truffle** - uses the [Truffle Flattener](https://www.npmjs.com/package/truffle-flattener)
+* **Hardhat** - uses the [Hardhat flattener](https://hardhat.org/hardhat-runner/docs/advanced/flattening) 
+* The [POA Solidity Flattener](https://github.com/poanetwork/solidity-flattener) can be used to flatten a smart contract
+* **Remix** - provides a built-in flattener tool
+ 
+**Note:** In the **Remix Explorer**, select your smart contract file ( **.sol** ) to flatten, then right click to the flattener tool.
```
Comment (taijonsmith) [inline docs/deploy.md:592]: is this supposed to be `right click on the flattener tool.`?

File: docs/deploy.md [hunk 4]
```diff
@@ -548,3 +548,102 @@ This exposes a JSON-RPC interface to the Hardhat Network. Use the address, **htt
...
+
+![alt_text](/img/docs/deploy/viaflattenedsource.png)
+
+6. In the **New Smart Contract Verification** page, click **Next**. A form page appears. 
+
+![alt_text](/img/docs/deploy/verifyform.png)
+
+Set the form with the following:
+
+  * **Contract name:** The name of the class called in the **.sol** file is the contract name. For example, in contract Owner {.., Owner is the contract name.
```
Comment (taijonsmith) [inline docs/deploy.md:622]: is `Owner {...` missing a closing brace?

File: docs/deploy.md (hunk 3 above)
Comment (Ramone-HL) [reply docs/deploy.md]: corrected

File: docs/deploy.md (hunk 4 above)
Comment (Ramone-HL) [reply docs/deploy.md]: corrected

File: static/img/docs/deploy/verifyssuccess1.png
Comment (Ramone-HL) [reply static/img/docs/deploy/verifyssuccess1.png:1]: nice

**Comment by yurikoinaba** [comment]
This is the error in Cloudflare pages:
```
12:39:14.821 | Error: Image static/img/docs/deploy/verifysuccess1.png used in docs/deploy.md not found.
-- | --
12:39:14.821 | at async Promise.all (index 5)
12:39:14.821 | [ERROR] Client bundle compiled with errors therefore further build is impossible.
```

File: docs/deploy.md [hunk 5]
```diff
@@ -548,3 +548,102 @@ This exposes a JSON-RPC interface to the Hardhat Network. Use the address, **htt
...
+New RPC URL:  https://yuma-testnet.horizenlabs.io/ethv1 
+Chain ID: 1662 
+Currency symbol: ZEN
+```
+
+In this exercise, the [EON Explorer](https://yuma-explorer.horizen.io/) is connected to the Yuma Testnet. The smart contract was developed and deployed to the Yuma Testnet network. The verification process will be performed using the **Via flattened source code** option in the EON Explorer.
+
+### Flatten the Smart Contract
+
+Before using the EON Explorer, the smart contract you want to verify has been deployed in the Yuma Testnet network. You need to flatten the smart contract if the contract inherits from other contracts or if it uses a library. Flattening a smart contract refers to combining all Solidity code into one file instead of multiple source files, thereby having the imported code be embedded in the same file.
```
Comment (taijonsmith) [inline docs/deploy.md:583]: seems like a word is missing in this first sentence

File: docs/deploy.md [hunk 6]
```diff
@@ -548,3 +548,102 @@ This exposes a JSON-RPC interface to the Hardhat Network. Use the address, **htt
...
+In this exercise, the [EON Explorer](https://yuma-explorer.horizen.io/) is connected to the Yuma Testnet. The smart contract was developed and deployed to the Yuma Testnet network. The verification process will be performed using the **Via flattened source code** option in the EON Explorer.
+
+### Flatten the Smart Contract
+
+Before using the EON Explorer, the smart contract you want to verify has been deployed in the Yuma Testnet network. You need to flatten the smart contract if the contract inherits from other contracts or if it uses a library. Flattening a smart contract refers to combining all Solidity code into one file instead of multiple source files, thereby having the imported code be embedded in the same file. 
+
+All smart contract development tools have a source code flattener. For example: 
+
+* **Truffle** - uses the [Truffle Flattener](https://www.npmjs.com/package/truffle-flattener)
+* **Hardhat** - uses the [Hardhat flattener](https://hardhat.org/hardhat-runner/docs/advanced/flattening)
```
Comment (taijonsmith) [inline docs/deploy.md:588]:
```suggestion
* **Hardhat** - uses the [Hardhat Flattener](https://hardhat.org/hardhat-runner/docs/advanced/flattening) 
```
small, but for consistency

File: docs/deploy.md [hunk 7]
```diff
@@ -548,3 +548,102 @@ This exposes a JSON-RPC interface to the Hardhat Network. Use the address, **htt
...
+Before using the EON Explorer, the smart contract you want to verify has been deployed in the Yuma Testnet network. You need to flatten the smart contract if the contract inherits from other contracts or if it uses a library. Flattening a smart contract refers to combining all Solidity code into one file instead of multiple source files, thereby having the imported code be embedded in the same file. 
+
+All smart contract development tools have a source code flattener. For example: 
+
+* **Truffle** - uses the [Truffle Flattener](https://www.npmjs.com/package/truffle-flattener)
+* **Hardhat** - uses the [Hardhat flattener](https://hardhat.org/hardhat-runner/docs/advanced/flattening) 
+* The [POA Solidity Flattener](https://github.com/poanetwork/solidity-flattener) can be used to flatten a smart contract
+* **Remix** - provides a built-in flattener tool
+ 
+**Note:** In the **Remix Explorer**, select your smart contract file ( **.sol** ) to flatten, then right-click to the flattener tool.
```
Comment (taijonsmith) [inline docs/deploy.md:592]:
```suggestion
**Note:** In the **Remix Explorer**, select your smart contract file ( **.sol** ) to flatten, then right-click on the flattener tool. 
```
I'm not sure if "right-click **to** the flattener tool" is grammatically correct. Can you confirm? Or maybe it should be "right-click to access the flattener tool". I'm not sure of the context

File: docs/deploy.md [hunk 8]
```diff
@@ -548,3 +548,102 @@ This exposes a JSON-RPC interface to the Hardhat Network. Use the address, **htt
...
+
+6. In the **New Smart Contract Verification** page, click **Next**. A form page appears. 
+
+![alt_text](/img/docs/deploy/verifyform.png)
+
+Set the form with the following:
+
+  * **Contract name:** The name of the class called in the **.sol** file is the contract name. For example, *Owner* is the contract name.
+
+  * **Include Nightly Builds:** By default, **No** is populated. Enter **Yes** to display nightly builds.
```
Comment (taijonsmith) [inline docs/deploy.md:624]:
```suggestion
  * **Include Nightly Builds:** By default, **No** is selected. Enter **Yes** to display nightly builds.
```

File: docs/deploy.md [hunk 9]
```diff
@@ -548,3 +548,102 @@ This exposes a JSON-RPC interface to the Hardhat Network. Use the address, **htt
...
+
+  * **Compiler:** Make sure that you select the same version that you had compiled your contract.
```
Comment (taijonsmith) [inline docs/deploy.md:626]:
```suggestion
  * **Compiler:** Make sure that you select the same version that you used to compile your contract.
```

File: docs/deploy.md [hunk 10]
```diff
@@ -548,3 +548,102 @@ This exposes a JSON-RPC interface to the Hardhat Network. Use the address, **htt
...
+
+  * **Enter the Solidity Contract Code:** Enter your Solidity contract code in this field. For example, the content of the **Token.sol** file is entered. 
+
+  * **Try to fetch constructor arguments automatically:** By default, **Yes** is enabled. If similar contracts exist, their constructor arguments may be available.
+
+  * **ABI-encoded Constructor Arguments:** See the [ABI Constructor Arguments](https://docs.blockscout.com/for-users/abi-encoded-constructor-arguments) for detailed information. 
+
+  * **Add Contract Libraries:** Enter the **name** and **0x address** of any required libraries called in the **.sol** file. If your contract is  flattened, you do not need to add a contract library.
+
+7. Click **Verify & publish**. A checkmark symbol appears next in the **Code** tab, if successful. Also, the **Read Contract** tab and the **Write Contract** tab appears.
```
Comment (taijonsmith) [inline docs/deploy.md:640]:
```suggestion
7. Click **Verify & publish**. A checkmark symbol appears next to the **Code** tab if successful. Also, the **Read Contract** tab and the **Write Contract** tab will appear.
```
Again, check that the context is correct here

**Review by taijonsmith — APPROVED** [review]

File: docs/deploy.md [hunk 11]
```diff
@@ -548,3 +548,102 @@ This exposes a JSON-RPC interface to the Hardhat Network. Use the address, **htt
...
+* The [POA Solidity Flattener](https://github.com/poanetwork/solidity-flattener) can be used to flatten a smart contract
+* **Remix** - provides a built-in flattener tool
+ 
+**Note:** In the **Remix Explorer**, select your smart contract file ( **.sol** ) to flatten, then right-click on the flattener tool. 
+
+### Check for the Deployed Contract
+
+In the [EON Explorer](https://yuma-explorer.horizen.io/), perform the following steps to begin the verification process:
+
+1. Once you have deployed your contract, your developer tool will display a contract address to check the pending transaction. Copy the contract address.
```
Comment (voyenavion) [inline docs/deploy.md:598]: "check the pending transaction" is not correct. if the contract address has been returned that means the transaction is confirmed. You could say instead: "Once you have deployed your contract, your developer tool will display the address the network has assigned your contract to. Copy that address."

**Review by voyenavion — CHANGES_REQUESTED** [review]

**Review by voyenavion — APPROVED** [review]
Looks great!

**Pull request merged**